### PR TITLE
Added create_tenant() and update_tenant() APIs

### PR DESCRIPTION
**Pull request by hiranya911**

### CODE REVIEW

File: firebase_admin/tenant_mgt.py [hunk 1]
```diff
@@ -141,6 +194,56 @@ def get_tenant(self, tenant_id):
...
+        except requests.exceptions.RequestException as error:
+            raise _auth_utils.handle_auth_backend_error(error)
+        else:
+            return Tenant(body)
+
+    def update_tenant(
+            self, tenant_id, display_name=None, allow_password_sign_up=None,
+            enable_email_link_sign_in=None):
+        """Updates the specified tenant with the given parameters."""
+        if not isinstance(tenant_id, str) or not tenant_id:
```
Comment (lahirumaramba) [inline firebase_admin/tenant_mgt.py]: I think `isinstance(tenant_id, str)` covers `not tenant_id` cases, right?
Also, did we choose not to use `_auth_utils.validate_string` here to provide a more specific error message?

**Comment by lahirumaramba** [comment]
@hiranya911 should we get the docs reviewed by someone or would you rather do that in the final merge PR?

File: firebase_admin/tenant_mgt.py (hunk 1 above)
Comment (hiranya911) [reply firebase_admin/tenant_mgt.py]: That doesn't cover the case where `tenant_id == ''`.

It's the same reason why we can't use `validate_string()` here.

**Comment by hiranya911** [comment]
I'll get the docs reviewed towards the end, when everything's in place.

**Pull request merged**